enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+In this sense, Bundle is a generalization of the image concept.
+
+The BundleDeployment API is analogous to the core Deployment Kubernetes resource, and allows users to specify a Bundle template, and that template contains the desired specification
+of a Bundle resource. The management of a Bundle and BundleDeployment resources is delegated to individual "provisioner" implementations.
+
+Both of these APIs are cluster-scoped resources.
+
+### Provisioner
+
+A "provisioner" is a controller that understands the Bundle and BundleDeployment APIs. Individual provisioner implementations use ingress-style "class" name identifiers to distinguish between the BundleDeployments and Bundles resources that it needs to manage. Each provisioner implementation requires the definition of a unique provisioner IDs.
+This provisioner pattern helps abstract the variations of bundle formats that are defined throughout the Kubernetes landscape. Future provisioner implementations may manage helm charts, crossplane packages, and OLM bundle formats.
+
+Both the Bundle and BundleDeployment APIs require a unique provisioner ID to be specified. This enables higher level systems (e.g. a PO manager) to generate BundleDeployments that referencing different unique provisioner IDs. This is useful for components that want more control over how to source, and apply, Bundle contents.
+
+## Proposal
+
+### Introducing a "PlatformOperator" CRD
+
+This first phase will introduce a new, top-level "PlatformOperator" cluster-scoped CRD. While admins today could include OLM manifests before cluster installation, this approach lacks the ability to influence the cluster's lifecycle.
+This new CRD will bring CVO awareness of these manifests, and help pave the way for adding a first-class UX (e.g. via the installer's install-config API) for configuring POs in future phases.
+
+Given this is an openshift-only API, this proposal recommends introducing the CRD into the openshift/api repository under the `platform.openshift.io/v1alpha1` API group. During phase 0, this resource will only contain a single, required `spec.packageName` field. This field will be used to configure the desired OLM-based Operator package name.
+Future phase requirements may necessitate more granular admin controls over how upgrades are approved, and how PO packages are sourced from OLM catalog content.
+
+Before cluster creation, an admin would be responsible for creating, and editing, the cluster's Kubernetes manifests with their desired PlatformOperator(s) configuration, and a FeatureGate manifest that enables the "TechPreviewNoUpgrades" feature set. An example for how to configure an individual PO resource, and how to enable POs through this FeatureGate manifest will be detailed below.
```

Comment:
This sentence if the first mention in this document of Tech Preview/`TechPreviewNoUpgrade`, we should probably mention explicitly that phase 0 is planned to be a tech preview before we talk about users having to set up a FeatureGate

Reply:
I missed that note! Never mind, I do think mentioning in the goals/non-goals has some value though

Reply:
Added "Admins can enable this PO mechanism in "TechPreviewNoUpgrade" enabled clusters." as a goal in the latest commit.